game/state/city/vequipment.cpp

```diff
@@ -67,12 +69,22 @@ bool VEquipment::fire(GameState &state, Vec3<float> targetPosition, Vec3<float>
 	}
 	this->reloadTime = type->fire_delay;
 	this->weaponState = WeaponState::Reloading;
+
+	number_of_shots = this->type->burst;
+	if (number_of_shots == 0)
```

Comment:
Would this be easier to set the 'default' burst to 1?

Reply:
That is what i initially changed from Andrey's code. HOWEVER, because many weapons do not have a <burst> value set, it defaults to null/0 causing a CTD unless i explicitly state that a null/0 value is actually a single shot.

Additionally, this protects the game from bad modding etiquette where modders may, instead of just removing <burst> from the XML just simply state "0" as the value out of laziness or misunderstanding

It's a little over-engineered, but at least the above method is largely foolproof with regards to avoiding problems further down the line; most of which will be generated by any modding community.